### ISSUE TITLE
Cannot jump to macro-generated definitions

### ISSUE THREAD

**Issue by gabro** [scalameta/metals#101]
Example

Use [`enumero`](https://github.com/buildo/enumero) to define an enum as:

```scala
@enum trait Planet {
  object Earth
  object Venus
  object Mercury
}
```

The generated code is something like

```scala
sealed abstract trait Planet
object Planet {
  case object Earth extends Planet
  case object Venus extends Planet
  case object Mercury extends Planet
}
```

When using it, the definition provider finds nothing:

```scala
val x = Planet.Earth // no definition found for `Earth`
```

This is not super-obvious to fix, since the positions depend on the macro implementation.
@olafurpg suggested we may run the syntactic indexer (`ScalaMtags`) to fill-in positions that are not present in the semanticdb.

**Comment by olafurpg** [comment]
#139 could offer a solution for this: combine syntactic and semantic indexing of workspace files. I suspect the positions from the expanded macro are messing up semanticdb-scalac so it doesn't pick up the definition position. If we index all *.scala files in the workspace with mtags then we will pick up the definition positions.

**Comment by olafurpg** [comment]
Fixed in #332 and test case to assert macro expanded classes can be jumped to 

https://github.com/scalameta/metals/blob/0c3aef07d60128a94c52b15ebdf78b242b9fdc16/tests/unit/src/test/resources/definition/example/Definitions.scala#L13
